### PR TITLE
Change format of project to conform to resin standards

### PR DESCRIPTION
**Pull request by CameronDiver**
* Use build directory
* Use tsconfig include
* Do in memory test compilation

Change-type: patch
Signed-off-by: Cameron Diver <cameron@resin.io>

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-    "outDir": "./lib/",
+    "outDir": "./build/",
     "sourceMap": true,
     "types": ["node"],
     "target": "es6",
     "noImplicitAny": true,
     "strictNullChecks": true
   },
+  "include": [
+    "src/"
```
Comment (hedss) [inline tsconfig.json]: I *think* this needs to be explicit, else `tsc` won't follow it, eg.:
```
"include": [
  "src/**/*.ts"
]
```

Comment (CameronDiver) [inline tsconfig.json]: It seems to be working both on this module and `resin-bundle-resolve`, e.g. it definitely compiles in all files etc.

File: package.json [hunk 2]
```diff
@@ -2,13 +2,11 @@
...
   "description": "A containerised builder which interacts with the docker remote API to perform builds.",
-  "main": "lib/index.js",
+  "main": "build/index.js",
   "types": "src/index.ts",
   "scripts": {
-    "start": "",
     "build": "gulp",
-    "pretest": "tsc tests/tests.ts",
-    "test": "mocha tests/tests.js"
+		"test": "gulp test"
```
Comment (pimterry) [inline package.json]: Indentation

Comment (CameronDiver) [inline package.json]: Vim hates package.json files...

**Pull request merged**